### PR TITLE
Add possibility to post content via Http class

### LINKED ISSUE #3325: Support for POST in Piwik_Http::sendHttpRequest
Piwik_Http::sendHttpRequest does not currently support [the POST method](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.5).

Required by #2708 to go around a Clockwork limitation.

### PR DESCRIPTION
**Pull request by tsteur**
fixes #3325
Needed for premium plugins since I have to post an accessToken to the Marketplace API as we do not want to have the token in the URL and the webserver logs.

Tested it manually with all 3 methods (socket, fopen and curl) and always worked. Also worked with an HTTPS url. 

Note: We cannot add a parameter for this to `Http::sendHttpRequest` method since it would break API. (Other plugins may extend this class and overwrite methods similar to https://github.com/piwik/piwik/issues/9531 )

### CODE REVIEW

**Comment by tsteur** [comment]
I can add automated tests but we'd need a domain that accepts a post request and returns the posted content or similar to actually test it.

**Comment by sgiehl** [comment]
For tests we can use local files like done in the Authentication tests. See https://github.com/piwik/piwik/blob/master/tests/PHPUnit/Integration/HttpTest.php#L152

**Comment by tsteur** [comment]
@sgiehl Thx I'll have a look

**Comment by tsteur** [comment]
done

File: tests/PHPUnit/Integration/Http/Post.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+// used in integration tests to see if POST method works.
+// for security reasons we allow max 3 post vars, each key and value is only allowed to have max 6 hex characters
+
+function accept($key)
```
Comment (tsteur) [inline tests/PHPUnit/Integration/Http/Post.php]: If someone installs Piwik via Git on production this file will be on a production server. Therefore I added a little check to make sure people can't do too many things with it

**Comment by tsteur** [comment]
Actually, this could be even merged in 2.16.0 as it is unlikely to break anything.

**Comment by mattab** [comment]
Looks good :+1:

**Pull request merged**